libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp

```diff
@@ -374,7 +374,8 @@ uint8_t ESP8266WiFiSTAClass::waitForConnectResult() {
     if((wifi_get_opmode() & 1) == 0) {
         return WL_DISCONNECTED;
     }
-    while(status() == WL_DISCONNECTED) {
+    int i = 0;
+    while((!status() || status() >= WL_DISCONNECTED) && i++ < 100) {

```

Comment:
This is a max of 10s time. Some connections could take longer, especially under poor signal conditions. Some users could not want to spend so much time trying to connect and want to give up sooner. I suggest taking an argument that defaults to this when not set.

Reply:
What would happen with a default non inifinite value when user's network connectivity / the router is for any reason powered down or disconnected for a while (a longer while than this default value), or takes long to power on, or whatever ?

My point is that usually sketches patiently wait for connectivity. With this change, all sketches would need to be updated to loop on the connectivity loop.

I don't question this PR, only the historical default behaviour / backward compatibility.

Reply:
Oops @d-a-v is correct: either we maintain backwards compatibility and merge now, or we keep this breaking change and wait until release 3.0.0 to merge. I prefer the former.
So: 
1. the function should take a timeout argument, in ms, which defaults to some value that disables the timeout (std::numeric_limits?)
2. If the user wants a timeout, he can specify a value for the argument.
3. In a future release we'll change the default to some value like 10 or 15s.

@ShinDaiIchi Could you please implement the changes and update the PR?